src/mindlessgen/qm/xtb.py

```diff
@@ -34,7 +34,20 @@ def optimize(
         """
         Optimize a molecule using xtb.
         """
-
+        nel = 0
+        for atom in molecule.ati:
+            nel += atom + 1
+        # Check if the number of the remaning electrons is even
+        test_rhf = nel - molecule.uhf - molecule.charge
```

Comment:
Is this line supposed to emulate that you subtract the number of `f` electrons? (`- molecule.uhf`)

Reply:
Or atleast the number of unpaired electrons

Reply:
But shouldn't you then subtract the total number of `f` electrons in the 3+ case? (Wouldn't make a difference regarding unpaired electrons anyway)

Reply:
This would only make a difference for the elements Tb to Lu, but if we want to be exact, this should also be subtracted right.